src/auth/auth-provider.tsx

```diff
@@ -3,7 +3,9 @@ import type { AuthorizationPayload } from '@serlo/authorization'
 import { createContext, ReactNode, useEffect, useState } from 'react'
 
 import { AuthSessionCookie } from './cookie/auth-session-cookie'
+import { fetchAndPersistAuthSession } from './cookie/fetch-and-persist-auth-session'
```

Comment:
@elbotho I have overlooked the bundle size analyzer.

Including this function will load the whole kratos client for every route.

We need to lazy load kratos here ... I would suggest an `import()` here:

https://nextjs.org/docs/advanced-features/dynamic-import#with-external-libraries

Reply:
Hmm I see. 
The trouble is, that we'd actually use the call on all routes.
At least for probably logged in users (and on staging for now).

Reply:
Dynamic import should exclude it from first load JS, which will keep hydration fast and defer it a bit.

It's ok that it's used on all routes. As far as I understood it's only loaded for users that have an active session? Only these users will load the additional JS.

Reply:
nice, if everything goes well it should not even cause additional js loading, since kratos is needed for logging in / out anyway.